core/include/traccc/fitting/kalman_filter/kalman_actor.hpp

```diff
@@ -109,22 +135,31 @@ struct kalman_actor : detray::actor {
             // Run Kalman Gain Updater
             const auto sf = navigation.get_surface();
 
-            const bool res =
-                sf.template visit_mask<gain_matrix_updater<algebra_t>>(
+            bool res = false;
+
+            // Forward filter
+            if (!actor_state.backward_mode) {
```

Comment:
Is the whole counting needed again during smoothing or does it need to be moved into this if-block?

Reply:
Sorry, what is the counting?

Reply:
A sorry, typo! I meant hole counting (l.127)

Reply:
Fixed